include/vulkan/layer/vk_layer_settings.h

```diff
@@ -21,44 +21,23 @@
 
 #pragma once
 
-#include <string>
-#include <vector>
+#ifdef __cplusplus
+extern "C" {
+#endif 
 
-#include <vulkan/vk_layer.h>
-#include <vulkan/vulkan.h>
+#include "vk_layer_settings_ext.h"
 
-namespace vku {
+typedef void *(*VL_LAYER_SETTING_LOG_CALLBACK)(const char *pSettingName, const char *pMessage);
 
-typedef std::vector<std::string> Strings;
-typedef std::vector<std::pair<std::string, int>> List;
-typedef void *(*LAYER_SETTING_LOG_CALLBACK)(const char *setting_key, const char *message);
+// Initialize the layer settings. If 'pCallback' is set to NULL, the messages are outputed to stderr.
+void vlInitLayerSettings(const char *pLayerName, const void *pNext, VL_LAYER_SETTING_LOG_CALLBACK pCallback);
```

Comment:
Instead of returning void couldn't this return VK_RESULT.

If given an invalid layer perhaps returning `VK_ERROR_LAYER_NOT_PRESENT` would make sense.

Reply:
I am also using VK_RESULT for `vlGetLayerSettingData` current investigation but yes that sounds like a good idea for `vlInitLayerSettings` too